cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -1711,22 +1713,21 @@ private void handleUnauthorizedResponse(Optional<ResponseBody> body, String resp
     /**
      * Handle unauthorized response.
      *
-     * @param body response body
+     * @param mediaType mediaType
+     * @param responseBody response body
      * @throws IOException            when accessing response body
      * @throws CentralClientException with unauthorized error message
      */
-    private void handleUnauthorizedResponse(Optional<ResponseBody> body)
+    private void handleUnauthorizedResponse(MediaType mediaType, String responseBody)
             throws IOException, CentralClientException {
-        if (body.isPresent()) {
-            Optional<MediaType> contentType = Optional.ofNullable(body.get().contentType());
-            if (contentType.isPresent() && isApplicationJsonContentType(contentType.get().toString())) {
-                Error error = new Gson().fromJson(body.get().string(), Error.class);
-                throw new CentralClientException("unauthorized access token. " +
-                        "check access token set in 'Settings.toml' file. reason: " + error.getMessage());
-            } else {
-                throw new CentralClientException("unauthorized access token. " +
-                        "check access token set in 'Settings.toml' file.");
-            }
+        Optional<MediaType> contentType = Optional.ofNullable(mediaType);
+        if (contentType.isPresent() && isApplicationJsonContentType(contentType.get().toString())) {
+            Error error = new Gson().fromJson(responseBody, Error.class);
+            throw new CentralClientException("unauthorized access token. " +
+                    "check access token set in 'Settings.toml' file. reason: " + error.getMessage());
+        } else {
+            throw new CentralClientException("unauthorized access token. " +
+                    "check access token set in 'Settings.toml' file.");
```

Comment:
Perhaps we can simplify this as,

```suggestion
        StringBuilder message = new StringBuilder("unauthorized access token. check access token set in 'Settings.toml' file.");
        if (contentType.isPresent() && isApplicationJsonContentType(contentType.get().toString())) {
            Error error = new Gson().fromJson(responseBody, Error.class);
            message.append(" reason: " + error.getMessage());
        }
        throw new CentralClientException(message.toString());
```

Even if we do it the way you suggested, we do not need the else clause AFAIU.

Reply:
Addressed with [98bed87](https://github.com/ballerina-platform/ballerina-lang/pull/42152/commits/98bed871eee6cbe9777641f0f5dd390cbacdde00)